test/TestAppUtils/VisualTreeDumper.cs

```diff
@@ -0,0 +1,229 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Media;
+
+namespace MUXControls.TestAppUtils
+{
+    class VisualTreeDumper
+    {
+        interface IVisitor
```

Comment:
Is abstracting out IVisitor/Visitor from the VisualTreeDumper getting us anything? Do we expect to have alternative implementations of IVisitor?

Reply:
It's just a habit from java programming, and It's easy to write a unit testing for it. But we don't have that kind of xunit testing project, so we didn't see benefit of it. I was expected to but actually I didn't see alternative implementation yet.  but I will not make change to the code

---
In reply to: [243671113](https://github.com/Microsoft/microsoft-ui-xaml/pull/126#discussion_r243671113) [](ancestors = 243671113)

Reply:
If you're going to keep the interfaces at least please comment them.
I'd also argue if there's no intention to have an alternate implementation then to just leave them out.

Reply:
removed interface